src/config/mongoose_config_parser.erl

```diff
@@ -200,3 +201,58 @@ add_dep_modules_opt({local_config, {modules, Host}, Modules}) ->
     {local_config, {modules, Host}, Modules2};
 add_dep_modules_opt(Other) ->
     Other.
+
+%% local functions
+-ifdef(TEST).
+halt_with_msg(ExitMsg, Errors) ->
+    config_error(ExitMsg, Errors).
+-else.
+halt_with_msg(ExitMsg, Errors) ->
+    [?LOG_ERROR(Error) || Error <- Errors],
+    mongoose_config_utils:exit_or_halt(ExitMsg).
```

Comment:
Please check it manually (if not done yet) by starting MongooseIM with some errors in the config file.

Reply:
well, it was crashing even test node. that's the reason to change implementation conditionally.

Reply:
I fully understand. What I meant was that these lines are moved so it would be good to make sure they still work. A sanity check.